### ISSUE TITLE
strange repeating sanitation

### ISSUE THREAD

**Issue by dgm** [pboling/sanitize_email#12]
I have a weird output in my sanitized emails:

```
Subject: (mortonda at dgrmm.net) (mortonda at dgrmm.net) (mortonda at
 dgrmm.net) (mortonda at dgrmm.net) (mortonda at dgrmm.net) (mortonda at
 dgrmm.net) (mortonda at dgrmm.net) (mortonda at dgrmm.net) (mortonda at
 dgrmm.net) (mortonda at dgrmm.net) (mortonda at dgrmm.net) (mortonda at
 dgrmm.net) (mortonda at dgrmm.net) (mortonda at dgrmm.net) (mortonda at
 dgrmm.net) (mortonda at dgrmm.net) (joe1 at example.com) Confirmation
 instructions
Mime-Version: 1.0
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit
X-Virus-Scanned: Maia Mailguard 1.0.2

: 
: 
: 
: 
: 
: 
: 
: 
: 
: 
: 
: 
: 
: 
: 
: 
: 
X-Sanitize-Email-To: joe1@example.com
X-Sanitize-Email-To: mortonda@dgrmm.net
X-Sanitize-Email-To: mortonda@dgrmm.net
X-Sanitize-Email-To: mortonda@dgrmm.net
X-Sanitize-Email-To: mortonda@dgrmm.net
X-Sanitize-Email-To: mortonda@dgrmm.net
X-Sanitize-Email-To: mortonda@dgrmm.net
X-Sanitize-Email-To: mortonda@dgrmm.net
X-Sanitize-Email-To: mortonda@dgrmm.net
X-Sanitize-Email-To: mortonda@dgrmm.net
X-Sanitize-Email-To: mortonda@dgrmm.net
X-Sanitize-Email-To: mortonda@dgrmm.net
X-Sanitize-Email-To: mortonda@dgrmm.net
X-Sanitize-Email-To: mortonda@dgrmm.net
X-Sanitize-Email-To: mortonda@dgrmm.net
X-Sanitize-Email-To: mortonda@dgrmm.net
X-Sanitize-Email-To: mortonda@dgrmm.net

<p>Welcome joe1@example.com!</p>
```

**Comment by gsar** [comment]
@pboling this issue is now happening consistently for me in my dev environment running pow and Application.config.cache_classes=false.

I think it has something to with with how the register_interceptor call is being made with an instance instead of a class/symbol.  At https://github.com/mikel/mail/blob/df48a05a7fb5a4271e6df12da7afb26a53494a18/lib/mail/mail.rb#L223 you can see the code handles ignoring duplicate symbols, but since we are passing an instance, that code won't detect dups.

In my dev environment, the code is being reloaded on every request, so there seem to be many instances getting registered, so you end up with repeating subject prefixes and x-sanitize-email-to headers as the number of requests goes up.

**Comment by pboling** [comment]
@gsar

**Comment by pboling** [comment]
The code in action mailer you linked to is constantizing the symbol/string passed in, or accepting whatever else is passed in without conversion.  So it normally registers a constant Class.  We should be registering a Class then, rather than an instance.  I will look into how that would affect functionality.

**Comment by gsar** [comment]
@pboling - yup, turning the instance method in our interceptor class into a class method and then passing a symbol into ActionMailer will most probably fix this.

**Comment by pboling** [comment]
@gsar - fix is proposed in #26

**Comment by pboling** [comment]
@gsar - fix released in v 1.2.0 to RubyGems.

**Comment by meetme2meat** [comment]
@pboling , @gsar  Just ran into this issue(with 1.1.7) because `delayed-job` keep [reload](https://github.com/collectiveidea/delayed_job/blob/master/lib/delayed/worker.rb#L187)(ing) the application in development mode causing the multiple `interceptor` to be registered to the `ActionMailer`.

Glad you fix it. 👍
